tests/test_field.py

```diff
@@ -124,6 +125,26 @@ def test_optional_redacting_of_missing_values(template_content, data, expected):
         ("((warning?))", "<span class='placeholder'>((warning?))</span>"),
         ("((warning? This is not a conditional))", "<span class='placeholder'>((warning? This is not a conditional))</span>"),
         ("((warning?? This is a warning))", "<span class='placeholder-conditional'>((warning??</span> This is a warning))"),
+        (
+            "((alert??With both (parenthesis) ))",
+            "<span class='placeholder-conditional'>((alert??</span>With both (parenthesis) ))",
+        ),
+        (
+            "((alert??Missing (right parenthesis ))",
+            "<span class='placeholder-conditional'>((alert??</span>Missing (right parenthesis ))",
+        ),
+        (
+            "((alert??Missing left parenthesis) ))",
+            "<span class='placeholder-conditional'>((alert??</span>Missing left parenthesis) ))",
+        ),
+        (
+            "((warning?)) and ((alert?? alert!))",
+            "<span class='placeholder'>((warning?))</span> and <span class='placeholder-conditional'>((alert??</span> alert!))",
+        ),
+        (
+            "(((warning))) and ((alert?? alert!))",
+            "(<span class='placeholder'>((warning))</span>) and <span class='placeholder-conditional'>((alert??</span> alert!))",
+        ),
```

Comment:
I added a couple of tests to make sure we are covered. The last test though make the current proposed changes. It passes though with the suggested added negative lookahead assertion.

Reply:
fixed!